### ISSUE TITLE
Blurry client with EGL

### ISSUE THREAD

**Issue by Janrupf** [gnif/LookingGlass#948]
## Description
Using the EGL renderer since B5 parts of the LG window appear blurry, even though window size and host resolution match.
Changing the renderer to OpenGL using `-g opengl` fixes the issue.

Window manager is i3-gaps, window size has been confirmed to be `1920x1080` using `xwininfo`.

Setting the resolution in Windows to `1680x1050` and forcing the looking glass client into this resolution using `wmctrl -i -r $window_id -e 0,1920,0,1680,1050` fixes the issue, however repeating the same steps for `1920x1080` yields a blurry result.

## Logs

`looking-glass-client` log:
```
[I]  19213802945              main.c:1304 | main                           | Looking Glass (B5.0.1)
[I]  19213802994              main.c:1305 | main                           | Locking Method: Atomic
[I]  19213815816           cpuinfo.c:36   | lgDebugCPU                     | CPU Model: AMD Ryzen Threadripper 1950X 16-Core Processor
[I]  19213815830           cpuinfo.c:37   | lgDebugCPU                     | CPU: 16 cores, 32 threads
[I]  19213842576              main.c:866  | lg_run                         | Using font: /usr/share/fonts/TTF/DejaVuSansMono.ttf
[I]  19213842626           ivshmem.c:128  | ivshmemOpenDev                 | KVMFR Device     : /dev/shm/looking-glass
[I]  19213868161               egl.c:261  | egl_initialize                 | Double buffering is off
[I]  19213868176              main.c:833  | tryRenderer                    | Using Renderer: EGL
[I]  19213869301               x11.c:227  | x11CheckEWMHSupport            | EWMH-complient window manager detected: i3
[I]  19213869569               x11.c:454  | x11Init                        | X11 XInput 2.0 in use
[I]  19213916396               x11.c:1374 | x11GetEGLDisplay               | Using eglGetPlatformDisplayEXT
[I]  19214304837               egl.c:696  | egl_renderStartup              | Multisampling enabled, max samples: 4
[I]  19214325441               egl.c:773  | egl_renderStartup              | Single buffer mode
[I]  19214384097               egl.c:789  | egl_renderStartup              | EGL     : 1.5
[I]  19214384114               egl.c:790  | egl_renderStartup              | Vendor  : NVIDIA Corporation
[I]  19214384116               egl.c:791  | egl_renderStartup              | Renderer: NVIDIA GeForce RTX 3080/PCIe/SSE2
[I]  19214384118               egl.c:792  | egl_renderStartup              | Version : OpenGL ES 3.2 NVIDIA 495.46
[I]  19214384128               egl.c:793  | egl_renderStartup              | EGL APIs: OpenGL_ES OpenGL
[I]  19214384133               egl.c:794  | egl_renderStartup              | EGL Exts: EGL_EXT_buffer_age EGL_EXT_client_sync EGL_EXT_create_context_robustness EGL_EXT_image_dma_buf_import EGL_EXT_image_dma_buf_import_modifiers EGL_MESA_image_dma_buf_export EGL_EXT_output_base EGL_EXT_stream_acquire_mode EGL_EXT_sync_reuse EGL_IMG_context_priority EGL_KHR_config_attribs EGL_KHR_create_context_no_error EGL_KHR_context_flush_control EGL_KHR_create_context EGL_KHR_fence_sync EGL_KHR_get_all_proc_addresses EGL_KHR_partial_update EGL_KHR_swap_buffers_with_damage EGL_KHR_no_config_context EGL_KHR_gl_colorspace EGL_KHR_gl_renderbuffer_image EGL_KHR_gl_texture_2D_image EGL_KHR_gl_texture_3D_image EGL_KHR_gl_texture_cubemap_image EGL_KHR_image EGL_KHR_image_base EGL_KHR_image_pixmap EGL_KHR_reusable_sync EGL_KHR_stream EGL_KHR_stream_attrib EGL_KHR_stream_consumer_gltexture EGL_KHR_stream_cross_process_fd EGL_KHR_stream_fifo EGL_KHR_stream_producer_eglsurface EGL_KHR_surfaceless_context EGL_KHR_wait_sync EGL_NV_nvrm_fence_sync EGL_NV_post_sub_buffer EGL_NV_quadruple_buffer EGL_NV_stream_consumer_eglimage EGL_NV_stream_cross_display EGL_NV_stream_cross_object EGL_NV_stream_cross_process EGL_NV_stream_cross_system EGL_NV_stream_dma EGL_NV_stream_flush EGL_NV_stream_metadata EGL_NV_stream_remote EGL_NV_stream_reset EGL_NV_stream_socket EGL_NV_stream_socket_inet EGL_NV_stream_socket_unix EGL_NV_stream_sync EGL_NV_stream_fifo_next EGL_NV_stream_fifo_synchronous EGL_NV_stream_consumer_gltexture_yuv EGL_NV_stream_attrib EGL_NV_stream_origin EGL_NV_system_time EGL_NV_output_drm_flip_event EGL_NV_triple_buffer EGL_NV_robustness_video_memory_purge EGL_WL_bind_wayland_display EGL_WL_wayland_eglstream
[I]  19214384143               egl.c:795  | egl_renderStartup              | GL Exts : GL_EXT_base_instance GL_EXT_blend_func_extended GL_EXT_blend_minmax GL_EXT_buffer_storage GL_EXT_clear_texture GL_EXT_clip_control GL_EXT_clip_cull_distance GL_EXT_color_buffer_float GL_EXT_color_buffer_half_float GL_EXT_conservative_depth GL_EXT_copy_image GL_EXT_depth_clamp GL_EXT_debug_label GL_EXT_discard_framebuffer GL_EXT_disjoint_timer_query GL_EXT_draw_buffers_indexed GL_EXT_draw_elements_base_vertex GL_EXT_EGL_image_array GL_EXT_EGL_image_storage GL_EXT_EGL_image_external_wrap_modes GL_EXT_float_blend GL_EXT_frag_depth GL_EXT_geometry_point_size GL_EXT_geometry_shader GL_EXT_gpu_shader5 GL_EXT_map_buffer_range GL_EXT_multi_draw_indirect GL_EXT_multisample_compatibility GL_EXT_multisampled_render_to_texture GL_EXT_multisampled_render_to_texture2 GL_EXT_multiview_texture_multisample GL_EXT_multiview_timer_query GL_EXT_occlusion_query_boolean GL_EXT_polygon_offset_clamp GL_EXT_post_depth_coverage GL_EXT_primitive_bounding_box GL_EXT_raster_multisample GL_EXT_render_snorm GL_EXT_robustness GL_EXT_separate_shader_objects GL_EXT_shader_group_vote GL_EXT_shader_implicit_conversions GL_EXT_shader_integer_mix GL_EXT_shader_io_blocks GL_EXT_shader_non_constant_global_initializers GL_EXT_shader_texture_lod GL_EXT_shadow_samplers GL_EXT_sparse_texture GL_EXT_sparse_texture2 GL_EXT_sRGB GL_EXT_sRGB_write_control GL_EXT_tessellation_point_size GL_EXT_tessellation_shader GL_EXT_texture_border_clamp GL_EXT_texture_buffer GL_EXT_texture_compression_bptc GL_EXT_texture_compression_dxt1 GL_EXT_texture_compression_rgtc GL_EXT_texture_compression_s3tc GL_EXT_texture_cube_map_array GL_EXT_texture_filter_anisotropic GL_EXT_texture_filter_minmax GL_EXT_texture_format_BGRA8888 GL_EXT_texture_mirror_clamp_to_edge GL_EXT_texture_norm16 GL_EXT_texture_query_lod GL_EXT_texture_rg GL_EXT_texture_shadow_lod GL_EXT_texture_sRGB_R8 GL_EXT_texture_sRGB_decode GL_EXT_texture_storage GL_EXT_texture_view GL_EXT_draw_transform_feedback GL_EXT_unpack_subimage GL_EXT_window_rectangles GL_KHR_context_flush_control GL_KHR_debug GL_EXT_memory_object GL_EXT_memory_object_fd GL_NV_memory_object_sparse GL_KHR_parallel_shader_compile GL_KHR_no_error GL_KHR_robust_buffer_access_behavior GL_KHR_robustness GL_EXT_semaphore GL_EXT_semaphore_fd GL_NV_timeline_semaphore GL_KHR_shader_subgroup GL_KHR_texture_compression_astc_ldr GL_KHR_texture_compression_astc_sliced_3d GL_KHR_texture_compression_astc_hdr GL_NV_bgr GL_NV_bindless_texture GL_NV_blend_equation_advanced GL_NV_blend_equation_advanced_coherent GL_NVX_blend_equation_advanced_multi_draw_buffers GL_NV_blend_minmax_factor GL_NV_clip_space_w_scaling GL_NV_compute_shader_derivatives GL_NV_conditional_render GL_NV_conservative_raster GL_NV_conservative_raster_pre_snap_triangles GL_NV_copy_buffer GL_NV_copy_image GL_NV_draw_buffers GL_NV_draw_instanced GL_NV_draw_texture GL_NV_draw_vulkan_image GL_NV_EGL_stream_consumer_external GL_NV_explicit_attrib_location GL_NV_fbo_color_attachments GL_NV_fill_rectangle GL_NV_fragment_coverage_to_color GL_NV_fragment_shader_barycentric GL_NV_fragment_shader_interlock GL_NV_framebuffer_blit GL_NV_framebuffer_mixed_samples GL_NV_framebuffer_multisample GL_NV_generate_mipmap_sRGB GL_NV_geometry_shader_passthrough GL_NV_instanced_arrays GL_NV_internalformat_sample_query GL_NV_gpu_shader5 GL_NV_image_formats GL_NV_memory_attachment GL_NV_mesh_shader GL_NV_occlusion_query_samples GL_NV_non_square_matrices GL_NV_pack_subimage GL_NV_packed_float GL_NV_packed_float_linear GL_NV_path_rendering GL_NV_path_rendering_shared_edge GL_NV_pixel_buffer_object GL_NV_polygon_mode GL_NV_primitive_shading_rate GL_NV_read_buffer GL_NV_read_depth GL_NV_read_depth_stencil GL_NV_read_stencil GL_NV_representative_fragment_test GL_NV_sample_locations GL_NV_sample_mask_override_coverage GL_NV_scissor_exclusive GL_NV_shader_atomic_fp16_vector GL_NV_shader_noperspective_interpolation GL_NV_shader_subgroup_partitioned GL_NV_shader_texture_footprint GL_NV_shading_rate_image GL_NV_shadow_samplers_array GL_NV_shadow_samplers_cube GL_NV_sRGB_formats GL_NV_stereo_view_rendering GL_NV_texture_array GL_NV_texture_barrier GL_NV_texture_border_clamp GL_NV_texture_compression_latc GL_NV_texture_compression_s3tc GL_NV_texture_compression_s3tc_update GL_NV_texture_dirty_tile_map GL_NV_timer_query GL_NV_viewport_array GL_NV_viewport_array2 GL_NV_viewport_swizzle GL_KHR_blend_equation_advanced GL_KHR_blend_equation_advanced_coherent GL_OES_compressed_ETC1_RGB8_texture GL_EXT_compressed_ETC1_RGB8_sub_texture GL_OES_depth24 GL_OES_depth32 GL_OES_depth_texture GL_OES_depth_texture_cube_map GL_OES_copy_image GL_OES_draw_buffers_indexed GL_OES_draw_elements_base_vertex GL_OES_texture_border_clamp GL_OES_tessellation_point_size GL_OES_tessellation_shader GL_OES_texture_buffer GL_OES_geometry_point_size GL_OES_geometry_shader GL_OES_gpu_shader5 GL_OES_shader_io_blocks GL_OES_texture_view GL_OES_primitive_bounding_box GL_OES_EGL_image GL_OES_EGL_image_external GL_OES_EGL_image_external_essl3 GL_OES_EGL_sync GL_OES_element_index_uint GL_OES_fbo_render_mipmap GL_OES_get_program_binary GL_OES_mapbuffer GL_OES_packed_depth_stencil GL_OES_rgb8_rgba8 GL_OES_sample_shading GL_OES_sample_variables GL_OES_shader_image_atomic GL_OES_shader_multisample_interpolation GL_OES_standard_derivatives GL_OES_surfaceless_context GL_OES_texture_cube_map_array GL_OES_texture_npot GL_OES_texture_float GL_OES_texture_float_linear GL_OES_texture_half_float GL_OES_texture_half_float_linear GL_OES_texture_stencil8 GL_OES_texture_storage_multisample_2d_array GL_OES_vertex_array_object GL_OES_vertex_half_float GL_OES_viewport_array GL_OVR_multiview GL_OVR_multiview2 GL_OVR_multiview_multisampled_render_to_texture GL_ANDROID_extension_pack_es31a 
[I]  19214384200               egl.c:858  | egl_renderStartup              | Debug messages disabled, enable with egl:debug=true
[I]  19214406992           eglutil.c:35   | swapWithDamageInit             | Using EGL_KHR_swap_buffers_with_damage
[I]  19214604615              main.c:1176 | lg_run                         | Host ready, reported version: B5.0.1
[I]  19214604638              main.c:1177 | lg_run                         | Starting session
[I]  19214822317              main.c:637  | main_frameThread               | Format: FRAME_TYPE_BGRA 1920x1080 stride:1920 pitch:7680 rotation:0
```
`looking-glass-host` log:
```
[I]   3889677397              time.c:85   | windowsSetTimerResolution      | System timer resolution: 488.2 μs
[I]   3889679979               app.c:544  | app_main                       | Looking Glass Host (B5.0.1)
[I]   3889682443           cpuinfo.c:36   | lgDebugCPU                     | CPU Model: AMD Ryzen Threadripper 1950X 16-Core Processor
[I]   3889685233           cpuinfo.c:37   | lgDebugCPU                     | CPU: 16 cores, 16 threads
[I]   3889688630           ivshmem.c:131  | ivshmemInit                    | IVSHMEM 0* on bus 0x3, device 0x1, function 0x0
[I]   3889695975               app.c:561  | app_main                       | IVSHMEM Size     : 32 MiB
[I]   3889696776               app.c:562  | app_main                       | IVSHMEM Address  : 0x2C10000
[I]   3889697602               app.c:563  | app_main                       | Max Pointer Size : 1024 KiB
[I]   3889698523               app.c:564  | app_main                       | KVMFR Version    : 14
[I]   3889701049               app.c:625  | app_main                       | Max Frame Size   : 14 MiB
[I]   3889701843               app.c:650  | app_main                       | Trying           : DXGI
[I]   3889706631              dxgi.c:322  | dxgi_init                      | Device Name       : \\.\DISPLAY2
[I]   3889707528              dxgi.c:323  | dxgi_init                      | Device Description: NVIDIA GeForce GTX 750 Ti
[I]   3889708525              dxgi.c:324  | dxgi_init                      | Device Vendor ID  : 0x10de
[I]   3889709332              dxgi.c:325  | dxgi_init                      | Device Device ID  : 0x1380
[I]   3889710082              dxgi.c:326  | dxgi_init                      | Device Video Mem  : 2007 MiB
[I]   3889710839              dxgi.c:327  | dxgi_init                      | Device Sys Mem    : 0 MiB
[I]   3889711566              dxgi.c:328  | dxgi_init                      | Shared Sys Mem    : 8190 MiB
[I]   3889832912              dxgi.c:435  | dxgi_init                      | Feature Level     : 0xb000
[I]   3889833383              dxgi.c:436  | dxgi_init                      | Capture Size      : 1920 x 1080
[I]   3889833841              dxgi.c:437  | dxgi_init                      | AcquireLock       : enabled
[I]   3889835429              dxgi.c:527  | dxgi_init                      | Source Format     : DXGI_FORMAT_B8G8R8A8_UNORM
[I]   3889836664               app.c:675  | app_main                       | Using            : DXGI
[I]   3889837172               app.c:676  | app_main                       | Capture Method   : Asynchronous
[I]   3890439321               app.c:350  | captureStart                   | ==== [ Capture Start ] ====
[I]   3890440430               app.c:290  | frameThread                    | Frame thread started
```

## Videos

https://user-images.githubusercontent.com/25827180/152032958-96bd56ad-31ca-49da-8333-bb1776de3ce0.mp4

(NOTE: Fullscreen the video/resize to 1920x1080 in order to see the issues, the blurry parts are **not** compression artifacts, the client really looks like this!)

**Comment by gnif** [comment]
As discussed on Discord, LG does not have any code that decides how to scale the image, we simply control how scaling happens IF the GPU decides it needs to scale the image to fit the view port.

As there is nothing in LG that can account for this behaviour (caused by external factors) I am closing this issue. Thanks for the detailed report though and coming into Discord to discuss the issue.

**Comment by Janrupf** [comment]
In case anyone stumbles upon this, this issue has been fixed by https://github.com/gnif/LookingGlass/commit/91b0cba1459257a045d38ac8a27cb36a7caafa69 which gnif pointed out to me on discord

**Comment by feisuzhu** [comment]
> In case anyone stumbles upon this, this issue has been fixed by [91b0cba](https://github.com/gnif/LookingGlass/commit/91b0cba1459257a045d38ac8a27cb36a7caafa69) which gnif pointed out to me on discord

Encountered this too, thankfully this is already fixed :)
